### PR TITLE
Real Estate Property Data #1003

### LINKED ISSUE #1003: Real Estate Property Data
# Aim

- To make a program to extract required information from web pages and put the data in well structured format using beautiful soup library in python.

# Details
-Take data from real estate websites and chart them in tabular form.

I would like to work on this issue under GSSoC'21, please assign it to me.

### PR DESCRIPTION
**Pull request by Himanshi2997**
# Description

To make a program to extract required information from web pages and put the data in well structured format using beautiful soup library in python.

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines(Clean Code) of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have created a helpful and easy to understand `README.md`
- [x] My documentation follows [`Template for README.md`](https://github.com/avinashkranjan/Amazing-Python-Scripts/blob/master/Template%20for%20README.md)
- [x] I have added the project meta data in the PR template.
- [x] I have created the ``requirements.txt`` file if needed.

## Project Metadata
If there is no-file/nothing to fill the below fields with, then type: none

Example: If no requirements.txt needed/present then type in Requirments none => Requirments: none

Category:

- [ ]  Calculators
- [ ]  AI/ML
- [x]  Scrappers
- [ ]  Social_Media
- [ ]  PDF
- [ ]  Image_Processing
- [ ]  Video_Processing
- [ ]  Games
- [ ]  Networking
- [ ]  OS_Utilities
- [ ]  Automation
- [ ]  Cryptography
- [ ]  Computer_Vision
- [ ]  Fun
- [ ]  Others

Title: Real Estate Webscrapper

Folder: Real Estate Webscrapper

Requirements: requirements.txt

Script: real_estate_webscrapper.py

Arguments: none

Contributor: Himanshi2997

Description: Take data from real estate websites and chart them in tabular form.

### CODE REVIEW

File: Real Estate Webscrapper/README.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+## Real Estate Webscrapper
+- It will take information from the real estate site and store it in the form of csv file making the data more organised and locally accessible.
+
+___
+
+### Requirements
```
Comment (kaustubhgupta) [inline Real Estate Webscrapper/README.md:6]:
```suggestion
## Requirements
```

File: Real Estate Webscrapper/README.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+- BeautifulSoup
+- Pandas
+---
+### How To install
```
Comment (kaustubhgupta) [inline Real Estate Webscrapper/README.md:10]:
```suggestion
## How To install
```

File: Real Estate Webscrapper/README.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+### Step 2
+- Use inspect in website to know which div contains the information that we need
+- Use beautiful soup to load the information in program and store it into a dictionary for each property
+
+### Step 3
+- Use pandas to convert the list of dictionaries to csv file
+---
+
+### Author
```
Comment (kaustubhgupta) [inline Real Estate Webscrapper/README.md:29]:
```suggestion
## Author
```

File: Real Estate Webscrapper/README.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+- Himanshi2997
```
Comment (kaustubhgupta) [inline Real Estate Webscrapper/README.md:30]:
```suggestion
[Himanshi2997](https://github.com/Himanshi2997)
```

File: Real Estate Webscrapper/README.md [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+---
+
+### Output
```
Comment (kaustubhgupta) [inline Real Estate Webscrapper/README.md:33]:
```suggestion
## Output
```

File: Real Estate Webscrapper/output2.csv [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+1,4.30Cr,15412/sqft,5 BHK,"Bahubali Enclave, Anand Vihar ",2790 sqft
+2,1.23Cr,,2 BHK,"Goodwill Apartment, Rohi Sector 13 ",79 sqyrd
+3,1.10Cr,11579,2 BHK,Rohi Sector 9 ,950 sqft
+4,2.10Cr,9545/sqft,4 BHK,"Belur Apartments, Dwarka Sector 18A ",2000 sqft
+5,39Lac,,2 BHK,"Vishwakarma Colony, Mehrauli Badarpur Road ",500 sqft
+6,4.25Cr,17708/sqft,3 BHK,Safdarjung Enclave ,2300 sqft
+7,38Lac,5135,1 BHK,"Knowledge Park3,greater Noida ",740 sqft
+8,3.50Cr,25000/sqft,3 BHK,Sarvodaya Enclave ,1400 sqft
+9,7Cr,21440/sqft,4 BHK,"DLF Capital Greens, Moti Nagar, Kirti Nagar ",3265 sqft
+10,1Cr,,2 BHK,"Geetanjali Enclave, Malviya Nagar ",500 sqft
```
Comment (kaustubhgupta) [inline Real Estate Webscrapper/output2.csv:12]: delete this output file

File: Real Estate Webscrapper/real_estate_webscrapper.py [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+c=r.content
+soup=BeautifulSoup(c,"html.parser")
+
+
+l=[]
+
+
+all=soup.find_all("div",{"class":"flex relative clearfix m-srp-card__container"})
+for item in all:
+    d={}
```
Comment (kaustubhgupta) [inline Real Estate Webscrapper/real_estate_webscrapper.py:20]: can we use a better variable name that can explain the code better?

File: Real Estate Webscrapper/real_estate_webscrapper.py [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+
+    
+    title=item.find("span",{"class":"m-srp-card__title"})
+    
+    words=(title.text.replace("in","")).split()
+        
+    for i in range(len(words)):
+        if words[i]=="sale" or words[i]=="Sale":            
+            break
+    s=""
```
Comment (kaustubhgupta) [inline Real Estate Webscrapper/real_estate_webscrapper.py:58]: can we use a better variable name that can explain the code better?

File: Real Estate Webscrapper/real_estate_webscrapper.py [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+    except:
+        d["Carpet Area"]=item.find("div",{"class":"luxury-srp-card__area__value"}).text
+    
+    
+    l.append(d)
+    
+    
+
+df=pandas.DataFrame(l)
+df.to_csv("output2.csv")
```
Comment (kaustubhgupta) [inline Real Estate Webscrapper/real_estate_webscrapper.py:75]:
```suggestion
df.to_csv("./Real Estate Webscrapper/scraped.csv")
```

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]

**Review by kaustubhgupta — APPROVED** [review]

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]
- add the requirements.txt file
- Update the PR template properly, adhere to the template provided. Check out approved PRs for better understanding

File: Real Estate Webscrapper/requirements.txt [hunk 10]
```diff
@@ -0,0 +1,7 @@
+Requirements
+  
+1. requests==2.25.1
+2. pandas==1.2.4
+3. beautifulsoup4==4.9.3
+4. pip==20.2.3
+5. python==3.9.4
```
Comment (kaustubhgupta) [inline Real Estate Webscrapper/requirements.txt:7]: this is not the way of defining the requirements file. Check out this file for ref: https://github.com/avinashkranjan/Amazing-Python-Scripts/blob/master/Amazon-Price-Alert/requirements.txt

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]

File: Real Estate Webscrapper/requirements.txt [hunk 11]
```diff
@@ -0,0 +1,5 @@
+requests==2.25.1
+pandas==1.2.4
+beautifulsoup4==4.9.3
+pip==20.2.3
+python==3.9.4
```
Comment (kaustubhgupta) [inline Real Estate Webscrapper/requirements.txt:5]: not valid requirements, remove them

**Review by kaustubhgupta — APPROVED** [review]

**Review by santushtisharma10 — APPROVED** [review]
LGTM

**Review by antrikshmisri — APPROVED** [review]
LGTM

**Review by avinashkranjan — APPROVED** [review]

**Pull request merged**